python/cudf/cudf/core/dataframe.py

```diff
@@ -3814,6 +3814,48 @@ def quantile(
             result.index = q
             return result
 
+    def quantiles(
+        self, q=0.5, interpolation="nearest",
+    ):
+        """
+        Return values at the given quantile.
+
+        Parameters
+        ----------
+
+        q : float or array-like
+            0 <= q <= 1, the quantile(s) to compute
+        interpolation : {`lower`, `higher`, `nearest`}
+            This parameter specifies the interpolation method to use,
+            when the desired quantile lies between two data points i and j.
+            Default 'nearest'.
+
+        Returns
+        -------
+
+        DataFrame
+
+        """
+        if isinstance(q, numbers.Number):
+            q_is_number = True
+            q = [float(q)]
+        elif isinstance(q, (list, tuple)):
+            q_is_number = False
+        else:
+            msg = "`q` must be either a single element or list"
+            raise TypeError(msg)
+
+        result = self._quantiles(q, interpolation.upper())
+
+        if q_is_number:
+            result = result.transpose()
```

Comment:
Does this work in the situation that we've got different dtype columns? Maybe we just shouldn't handle this case for now. Either way this is fine and we can revisit in a followup.